test/IntegrationTests/SmokeTests.cs

```diff
@@ -71,6 +71,31 @@ public void WhenClrProfilerIsNotEnabled()
 #endif
     }
 
+#if !NETFRAMEWORK
+    [Theory]
+    [Trait("Category", "EndToEnd")]
+    [InlineData(TestAppStartupMode.DotnetCLI)]
+    [InlineData(TestAppStartupMode.Exe)]
+    public void WhenClrProfilerIsNotEnabledStartupHookIsEnabledApplicationIsExcluded(TestAppStartupMode testAppStartupMode)
+    {
+        switch (testAppStartupMode)
+        {
+            case TestAppStartupMode.DotnetCLI:
+                SetEnvironmentVariable("OTEL_DOTNET_AUTO_EXCLUDE_PROCESSES", "dotnet,dotnet.exe");
+                break;
+            case TestAppStartupMode.Exe:
+                SetEnvironmentVariable("OTEL_DOTNET_AUTO_EXCLUDE_PROCESSES", $"{EnvironmentHelper.FullTestApplicationName},{EnvironmentHelper.FullTestApplicationName}.exe");
+                break;
+            default:
+                throw new ArgumentException($"{testAppStartupMode} is not supported by this test", nameof(testAppStartupMode));
+        }
+
+        SetEnvironmentVariable("CORECLR_ENABLE_PROFILING", "0");
```

Comment:
As I remember, by default profiling is disabled for .NET.

Reply:
I think it might be best to combine it into a single test with 2 params: `TestAppStartupMode` and `IsProfilerEnabled`.

Currently it also double reflects `ApplicationIsExcluded`

Reply:
I think that it is fine to either set that environment variable to "0" or to either "unset" that environment variable. Having that code in place can serve 2 purposes:

1. Make it clear that the native profiler is disabled.
2. "Defensively" protect the test code when the test runner accidentally inherits that environment variable from the environment.

Reply:
> by default profiling is disabled for .NET.

Yes, that is correct. However, as Chris noticed, the presence of the statement makes that clear that the test doesn't have CLR profiling.

@RassK I didn't want to combine the tests because the StartupHook doesn't apply to .NET Framework while the `ApplicationIsExcluded` applies to all runtimes.